### PR TITLE
feat: UI test for the flat fare journey by distance

### PR DESCRIPTION
**Pull request by NitishAroraKPMG**
## Description

UI test for the flat fare journey by distance

## Testing instructions

Run the flat fare cypress test.

### CODE REVIEW

File: repos/fdbt-site/cypress_tests/cypress/e2e/flatFare.cy.ts [hunk 1]
```diff
@@ -26,4 +26,14 @@ describe('The flat fare faretype product journey', () => {
         completeSalesPages();
         isFinished();
     });
+
+    it.only('completes successfully for pricing by distance', () => {
```
Comment (alainajaffer) [inline repos/fdbt-site/cypress_tests/cypress/e2e/flatFare.cy.ts:30]:
```suggestion
    it('completes successfully for pricing by distance', () => {
```

File: repos/fdbt-site/cypress_tests/cypress/support/helpers.ts [hunk 2]
```diff
@@ -297,6 +297,37 @@ export const selectRandomOptionFromDropDown = (dropDownId: string): void => {
...
+                const pricePerKm = ((i+1) * 3).toString();
+
+                cy.log(`Row(${i}): ${fromDistance}, ${toDistance}, Price: ${pricePerKm}`);
+                getElementById(`price-per-km-${i}`).type(pricePerKm);
+
+                if (i!=0) {                    
+                    getElementById(`distance-from-${i}`).clear().type(fromDistance);
+                }
+
+                if(i!==randomSelector-1) {
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/cypress_tests/cypress/support/helpers.ts:323]: linting doesnt look to be right here? 
```suggestion
                if (i !== randomSelector - 1) {
```

File: repos/fdbt-site/cypress_tests/cypress/support/steps.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { random } from 'cypress/types/lodash';
```
Comment (alainajaffer) [inline repos/fdbt-site/cypress_tests/cypress/support/steps.ts:1]: is this needed?

File: repos/fdbt-site/cypress_tests/cypress/support/helpers.ts [hunk 4]
```diff
@@ -297,6 +297,35 @@ export const selectRandomOptionFromDropDown = (dropDownId: string): void => {
...
+        }
+        cy.log(`Checking for ${i}`);
+        const fromDistance = (i * 2).toString();
+        const toDistance = ((i + 1) * 2).toString();
+        const pricePerKm = ((i + 1) * 3).toString();
+
+        cy.log(`Row(${i}): ${fromDistance}, ${toDistance}, Price: ${pricePerKm}`);
+        getElementById(`price-per-km-${i}`).type(pricePerKm);
+
+        if (i != 0) {
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/cypress_tests/cypress/support/helpers.ts:319]:
```suggestion
        if (i !== 0) {
```

**Review by alainajaffer — DISMISSED** [review]

**Review by alainajaffer — APPROVED** [review]

**Review by RobCattonKPMG — APPROVED** [review]

**Pull request merged**